### ISSUE TITLE
Save directly into pdf file instead of xopp

### ISSUE THREAD

**Issue by github2099** [xournalpp/xournalpp#4666]
**Is your feature request related to a problem? Please describe.**
- Not related to a problem -

**Describe the solution you'd like**
In this current stage, when opening a pdf and making changes, I am prompted to save in xopp format.

I hope that it will save directly into the pdf I am opening. 

**Describe alternatives you've considered**
- none -

**Additional context**
- none -

**Comment by GithubUser699** [comment]
You can just export the file directly as a PDF, e.g. by pressing `Control + e`.

You won't be able to edit any annotations previously done by you though.

**Comment by github2099** [comment]
True, but that's not very ideal

**Comment by rolandlo** [comment]
@github2099 What are your expectations? That you can save to PDF and edit the annotations as usual after reloading the PDF?

**Comment by github2099** [comment]
Yes true, I expect the annotation will be somehow saved on a "layer" on the PDF and I can edit afterward

**Comment by jhilmer** [comment]
Perhaps annotation could be saved on top in a pdf layer (perhaps they already are), and then attach the xopp as attachment to the pdf. Next time the pdf then is opened in xournal++ it could skip the pdf layer and rerender it based on the xopp attachment.

**Comment by TrevorShelton** [comment]
Maybe the PDF could have metadata that reminds Xournal++ of what was created using itself but doesn't affect the actual PDF or its usage.

**Comment by tigerjack** [comment]
I don't know if it's feasible, but [zotero](https://github.com/jlegewie/zotfile.git) has a [PDF toolkit](https://github.com/zotero/pdf-worker/tree/85d3c0ed13f128400e7e54243eacebf349b88fc5) which allows one to export the note attached to a PDF.

Basically, zotero allows one to annotate the PDFs inside the app, and then export the PDF with the notes.

I think some logic is common.

**Comment by ondrejnedoma** [comment]
This is honestly a big reason why I'm considering switching from xournal pp to something else for pdf annotations. The fact that I have to download the pdf, put it in a folder which I can never move (because it uses absolute paths for the pdf backgrounds), open the pdf in xournal pp, save it to an organised folder this time as xopp, and only then i can start annotations is quite annoying, and no other big pdf annotator does this as far as I know.

If other annotators can save **editable** pdf files, it's for sure possible to implement this.

**Comment by tigerjack** [comment]
I mostly agree @ondrejnedoma , but I think xournalpp was built to be more than a PDF annotator.

**Comment by chriswillmorris** [comment]
This would be wonderful to have!

**Comment by andrade** [comment]
My expectations:

1. Can save notes to xopp as is now
2. Can save notes directly to PDF instead of xopp
3. Can edit/remove notes later on when reopening PDF
4. Other PDF viewers can see the notes
5. Other PDF viewers can edit/remove the notes
6. Can edit/remove notes from other PDF editors
7. Show list of notes instead of table of contents

The behavior above is defintely possible. I use Xodo on Android that allows all these. And if I later open my PDF in Linux in evince I can edit the notes. I think Foxit also allowed it but I ditched it since no updates for many years. Some colleagues sometime send me PDFs with notes (they use different apps) and I can see the notes in my viewer, and edit them (although not necessarily handwritten notes). In the case of xournalpp, I opened an annotated PDF and couldn't interact with the notes.

4, 5 useful when I review a paper and send to a colleague. They use different PDF readers.

6 is useful when I receive an annotated paper from someone, with suggestions/fixes. I sometimes "tick" what I have done with the pen; or change the annotation color or add some text to existing comments justifying why I'm doing, or not doing, X (can't do this if I can't edit the annotations).

7 could be done with a toggle to show a _list of annotations_ instead of _table of contents_ (Evince does this) or maybe allow two columns side by side or one on top of the other (would be good in larger monitors).

Also useful, but not that important: Export an annotated PDF without the annotations. Use case is I annotate a scientific paper and colleague asks me for a copy so I could export a version without the annotations to mail to my colleague. If I'm revieweing a paper I want to send the annotations with the PDF, if it's personal notes then I don't want to send the annotations hence the option.
